### DISCUSSION TITLE
Procedure of registering image

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#542] [General]
TB 사용자가 cspImageId 를 알고 있는 경우:
1. `tumblebug/ns/$NSID/resources/image?action=registerWithId` API 를 사용
2. Tumblebug 이 Spider 에게 이미지 상세정보 요청
3. Spider 가 Tumblebug 에게 이미지 상세정보 응답
4. Spider 가 리턴한 이미지 상세정보를 이용하여 Tumblebug 이 TB image object 로 등록

---

'사용자가 cspImageId 를 모르고 있는 경우' 에 대해서도 Tumblebug 에서 고려가 필요할까요?

(webtool 에서의 procedure 예시: 
lookupImages에서 가져온 목록에서 특정 Image 선택하고 등록버튼을 누르면 해당 namespace에 image 저장)

**Comment by seokho-son** [comment]
@jihoon-seo 

> '사용자가 cspImageId 를 모르고 있는 경우' 에 대해서도 Tumblebug 에서 고려가 필요할까요?

이미 어느 정도 고려가 되어 있는 것이 아닐까요?
lookupImages 를 통해 cspImageId 를 구할 수 있고, image?action=registerWithId 를 통해 등록하면 되니까 지금도 처리는 가능한 것 같습니다. (다만.. 불편)

1. 운영자가 미리 모든 이미지들 다 등록해두어서 환경 설정급으로 처리하면, 사용자에게는 편하지 않을까 싶습니다.
2. default 또는 dynamic MCIR 개념이 어느정도 구현되면, 조금 더 편리해질 것 같습니다.
